### PR TITLE
feat: POAP manager contract implementation

### PR DESCRIPTION
**Pull request by dadamu**
This PR implements the POAP manager contract as described in [ADR-002](https://github.com/desmos-labs/desmos-contracts/blob/master/docs/architecture/adr-002-poap-manager-contract.md).

deps: #35

### CODE REVIEW

File: contracts/poap-manager/src/contract.rs
Comment (dadamu) [inline contracts/poap-manager/src/contract.rs:415]:
```suggestion
    fn mint_into_with_invalid_recipient_error() {
```

File: contracts/poap-manager/src/contract.rs
Comment (bragaz) [inline contracts/poap-manager/src/contract.rs:449]: Can we also add a test where a sender different than the admin try to update the admin?

File: contracts/poap-manager/src/error.rs
Comment (bragaz) [inline contracts/poap-manager/src/error.rs:16]:
```suggestion
    #[error("Instantiate POAP contract error")]
```

File: contracts/poap-manager/src/error.rs
Comment (bragaz) [inline contracts/poap-manager/src/error.rs:20]:
```suggestion
    NotAdmin {caller: Addr},
```

File: contracts/poap-manager/src/msg.rs
Comment (bragaz) [inline contracts/poap-manager/src/msg.rs:12]:
```suggestion
    /// Id of the POAP contract to be initialized along with this contract.
```

File: contracts/poap-manager/src/msg.rs
Comment (bragaz) [inline contracts/poap-manager/src/msg.rs:14]:
```suggestion
    /// Initialization message of the POAP contract.
```

File: contracts/poap-manager/src/msg.rs
Comment (bragaz) [inline contracts/poap-manager/src/msg.rs:56]:
```suggestion
    /// Address of the POAP contract
```

File: contracts/poap-manager/src/msg.rs
Comment (bragaz) [inline contracts/poap-manager/src/msg.rs:57]:
```suggestion
    pub poap_contract_address: Addr,
```

File: contracts/poap-manager/src/msg.rs
Comment (bragaz) [inline contracts/poap-manager/src/msg.rs:40]: Implementation is missing here

File: contracts/poap-manager/src/state.rs
Comment (bragaz) [inline contracts/poap-manager/src/state.rs:12]:
```suggestion
pub const POAP_CONTRACT_ADDRESS: Item<Addr> = Item::new("poap_contract_address");
```

File: contracts/poap-manager/src/msg.rs
Comment (dadamu) [reply contracts/poap-manager/src/msg.rs:40]: It is because the parameters of `ExecuteMsg` are all address string which should be verified by `deps.api.addr_validate(address)?`. I'll remove it since it does nothing.

**Comment by dadamu** [comment]
cargo clippy checking also checks the dependencies, even the non-features part. Should we remove the clippy checking from ci? @manu0466 @bragaz

File: .github/workflows/desmos_std.yml
Comment (dadamu) [inline .github/workflows/desmos_std.yml:90]: The error happens because of the unused features dependencies checking of `desmos-bindings` if we run `cargo clippy` on workplace for all contracts.

Comment (dadamu) [reply .github/workflows/desmos_std.yml:90]: The error messages is as follows, showing that unused features in the `mocks` feature:
```bash
error[E0432]: unresolved import `crate::posts::mocks`
  --> /home/kilem/.cargo/registry/src/github.com-1ecc6299db9ec823/desmos-bindings-1.0.2/src/mocks/mock_apps.rs:16:20
   |
16 | use crate::posts::{mocks::mock_posts_query_response, msg::PostsMsg};
   |                    ^^^^^ could not find `mocks` in `posts`

error[E0432]: unresolved import `crate::reactions::mocks`
  --> /home/kilem/.cargo/registry/src/github.com-1ecc6299db9ec823/desmos-bindings-1.0.2/src/mocks/mock_apps.rs:20:24
   |
20 | use crate::reactions::{mocks::mock_reactions_query_response, msg::ReactionsMsg};
   |                        ^^^^^ could not find `mocks` in `reactions`

error[E0432]: unresolved import `crate::relationships::mocks`
  --> /home/kilem/.cargo/registry/src/github.com-1ecc6299db9ec823/desmos-bindings-1.0.2/src/mocks/mock_apps.rs:22:28
   |
22 | use crate::relationships::{mocks::mock_relationships_query_response, msg::RelationshipsMsg};
   |                            ^^^^^ could not find `mocks` in `relationships`

error[E0432]: unresolved import `crate::reports::mocks`
  --> /home/kilem/.cargo/registry/src/github.com-1ecc6299db9ec823/desmos-bindings-1.0.2/src/mocks/mock_apps.rs:24:22
   |
24 | use crate::reports::{mocks::mock_reports_query_response, models::ReportTarget, msg::ReportsMsg};
   |                      ^^^^^ could not find `mocks` in `reports`

error[E0432]: unresolved import `crate::posts::mocks`
 --> /home/kilem/.cargo/registry/src/github.com-1ecc6299db9ec823/desmos-bindings-1.0.2/src/mocks/mock_queriers.rs:4:19
  |
4 | use crate::posts::mocks::mock_posts_query_response;
  |                   ^^^^^ could not find `mocks` in `posts`

error[E0432]: unresolved import `crate::reactions::mocks`
 --> /home/kilem/.cargo/registry/src/github.com-1ecc6299db9ec823/desmos-bindings-1.0.2/src/mocks/mock_queriers.rs:9:23
  |
9 | use crate::reactions::mocks::mock_reactions_query_response;
  |                       ^^^^^ could not find `mocks` in `reactions`

error[E0432]: unresolved import `crate::relationships::mocks`
  --> /home/kilem/.cargo/registry/src/github.com-1ecc6299db9ec823/desmos-bindings-1.0.2/src/mocks/mock_queriers.rs:11:27
   |
11 | use crate::relationships::mocks::mock_relationships_query_response;
   |                           ^^^^^ could not find `mocks` in `relationships`

error[E0432]: unresolved import `crate::reports::mocks`
  --> /home/kilem/.cargo/registry/src/github.com-1ecc6299db9ec823/desmos-bindings-1.0.2/src/mocks/mock_queriers.rs:13:21
   |
13 | use crate::reports::mocks::mock_reports_query_response;
   |                     ^^^^^ could not find `mocks` in `reports`
```

**Comment by bragaz** [comment]
@dadamu can you fix the conflicts a update the branch with master changes?

File: .github/workflows/desmos_std.yml
Comment (manu0466) [reply .github/workflows/desmos_std.yml:90]: This can be fixed if we use `#[cfg(not(target_arch = "wasm32"))]` instead of `#[cfg(all(not(target_arch = "wasm32"), ...))]` inside the `mod.rs` files of `desmos-bindings` like `cosmwasm-std`.
I think that we should go this way since can be annoying for another developer figuring out how to fix this error when runs `clippy`.

**Comment by dadamu** [comment]
@bragaz Fixed

**Comment by dadamu** [comment]
The `cargo clippy` checking will be fixed after https://github.com/desmos-labs/desmos-bindings/pull/37 merged and released

File: .github/workflows/desmos_std.yml
Comment (dadamu) [reply .github/workflows/desmos_std.yml]: @manu0466 Thanks so much, I didn't realize it is a bug from desmos-bindings.

**Review by bragaz — COMMENTED** [review]
Looks good to me, I'll approve this once we merged and fixed the clippy error within the bindings

**Comment by dadamu** [comment]
@bragaz Fixed

**Review by bragaz — APPROVED** [review]
Ready to go to me! 🚀

File: contracts/poap-manager/src/contract.rs
Comment (manu0466) [inline contracts/poap-manager/src/contract.rs:132]: I think we can have a function returning a simple bool.
Can be something like this:
``` rust
ProfilesQuerier::new(deps.querier.deref()).query_profile(user).is_ok()
```
Otherwise when `query_profile` fails because the user don't have a profile the call performed inside the `execute_claim` function never goes inside the if body and will never return `Err(ContractError::NoEligibilityError {})`

File: contracts/poap-manager/src/contract.rs
Comment (manu0466) [inline contracts/poap-manager/src/contract.rs:237]: We should avoid to have a custom serializer for an internal `cosmwasm-std` struct, if the cosmwasm team changes this struct our tests will probably start to fail. We should instead relay on the integration tests to check that the contract have the correct state with an integratino test.

File: contracts/poap-manager/src/contract.rs
Comment (manu0466) [inline contracts/poap-manager/src/contract.rs:389]: This can be removed since already covered in the `instantiate_propery` integration test.

**Review by manu0466 — APPROVED** [review]
Nice work!!

**Pull request merged**
